pkg/server/tls/dynamic_source.go

```diff
@@ -217,11 +218,16 @@ func (f *DynamicSource) Healthy() bool {
 
 func (f *DynamicSource) tryRegenerateCertificate(ctx context.Context, nextRenewCh chan<- time.Time) error {
 	return wait.PollUntilContextCancel(ctx, f.RetryInterval, true, func(ctx context.Context) (done bool, err error) {
+		f.log.Info("try to generate a serving certificate")
 		if err := f.regenerateCertificate(ctx, nextRenewCh); err != nil {
-			f.log.Error(err, "Failed to generate serving certificate, retrying...", "interval", f.RetryInterval)
+			if errors.Is(err, authority.ErrCertificateNotAvailable) {
+				f.log.Info(fmt.Sprintf("Certificate is not yet created. Will attempt again after %d seconds", f.RetryInterval/time.Second))
```

Comment:
```suggestion
				f.log.Info(fmt.Sprintf("Certificate is not yet created, retrying...", "interval", f.RetryInterval)
```

Why not use structural logging here, as in the log statement below? Is there a reason?

Reply:
Nope. I'll change the logging format.